File: demos/node/si1.js
import { GoogleGenAI } from "@google/genai";

const MODEL_NAME = "gemini-2.0-flash";
const API_KEY = process.env.GEMINI_API_KEY;

const ai = new GoogleGenAI({ apiKey: API_KEY });

async function processPrompt(prompt) {

  const response = await ai.models.generateContent({
    model: "gemini-2.0-flash",
    contents: prompt,
    config: {
      systemInstruction: `
You are a bot focused on astronomy topics, and only astronomy. For your answers, you should always try to involve
cats in some way. Refuse to answer questions outside of the topic. 
`
          },
  });


  return response.text;
}

if(process.argv.length < 3) {
  console.log('Pass a prompt argument... or else!');
  process.exit();
}

(async () => {

  let prompt = process.argv[2];
  console.log(`Generating response for prompt: ${prompt}`);
  let result = await processPrompt(prompt);
  console.log(result);

})();

File: demos/python/si1.py
from google import genai
from google.genai import types
import os 
import sys

if len(sys.argv) == 1:
	print("Pass a prompt argument... or else!")
	sys.exit()

prompt = sys.argv[1]

si = """
"You are a bot focused on astronomy topics, and only astronomy. For your answers, you should always try to involve
cats in some way. Refuse to answer questions outside of the topic. " 
"""

client = genai.Client(api_key=os.environ.get("GEMINI_API_KEY"))
response = client.models.generate_content(
    model="gemini-2.0-flash", contents=prompt,
	config=types.GenerateContentConfig(
		system_instruction=si
	)
)
print(response.text)